server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -940,6 +942,16 @@ private static void validateSnapshotRestorable(final String repository, final Sn
         }
     }
 
+    public static final Setting<Boolean> ALLOW_BWC_INDICES_SETTING = Setting.boolSetting(
+        "allow_bwc_indices",
+        false,
+        Setting.Property.NodeScope
+    );
+
+    private static boolean skipVersionChecks(RepositoryMetadata repositoryMetadata) {
+        return Build.CURRENT.isSnapshot() && ALLOW_BWC_INDICES_SETTING.get(repositoryMetadata.settings());
```

Comment:
we only allow this setting in snapshot builds. It is super rudimentary right now, but all we need to get the test running. This does not mean that suddenly everything will magically work with these older indices, as ES still has plenty of places where it checks `minimumIndexCompatibilityVersion`. This just gets the most basic blocks out of the way to get some tests running that allow us to work on the actual Lucene conversion logic.

Reply:
Could we actively reject the setting in real builds just to be on the safe side?

Reply:
sure, addressed in 1680afa